src/mds/MDCache.cc

```diff
@@ -7655,12 +7655,10 @@ int MDCache::path_traverse(MDRequestRef& mdr, Message *req, MDSInternalContextBa
   bool null_okay = (onfail == MDS_TRAVERSE_DISCOVERXLOCK);
   bool forward = (onfail == MDS_TRAVERSE_FORWARD);
 
-  assert(mdr || req || fin);
-  assert(!forward || mdr || req);  // forward requires a request
+  assert(mdr || req || fin || !forward);
 
   snapid_t snapid = CEPH_NOSNAP;
-  if (mdr)
-    mdr->snapid = snapid;
+  mdr->snapid = snapid;
```

Comment:
Why do you drop the sanity check here?

I'd suggest to add some sanity check at https://github.com/ceph/ceph/pull/9954/files#diff-2259d0a61fc0c9b2d2be3e40ade57303L7667 and https://github.com/ceph/ceph/pull/9954/files#diff-2259d0a61fc0c9b2d2be3e40ade57303L7667 ... instead...